### PR TITLE
Fix highlighting of apostrophes/ticks and qualified names

### PR DESCRIPTION
**Pull request by sheaf**
The main problem in highlighting of apostrophes was using the `\b` word boundary separator, which wouldn't match on an identifier ending with ticks because a tick is a non-word character (hence a lack of word boundary). I changed the word boundary pattern to the regex `(?![''\.\w])` which seems to work quite well. I also adjusted the regex for single-quoted characters to avoid conflicts with ticked identifiers.

Fixes the regression in #71.

**Edit:** Also fixed several issues with qualified names, adding test cases in `QualifiedNames.hs`:

qualified_infix

### CODE REVIEW

File: syntaxes/haskell.YAML-tmLanguage [hunk 1]
```diff
@@ -359,13 +360,15 @@ patterns:
     patterns:
       - include: '#type_signature'
     name: meta.type-declaration.haskell
-  - comment: >
+  - match: >-
+      ((?:[\p{Lu}\p{Lt}][\p{Ll}_\p{Lu}\p{Lt}\p{Nd}'']*\.)*)([\p{S}\p{P}&&[^(),;\[\]`{}_"']]+)
```
Comment (JustusAdam) [inline syntaxes/haskell.YAML-tmLanguage:364]: Do we have to add this to type operators as well?

File: syntaxes/haskell.YAML-tmLanguage [hunk 2]
```diff
@@ -520,12 +523,12 @@ repository:
...
     match: >-
-      \b[\p{Lu}\p{Lt}][\p{Ll}_\p{Lu}\p{Lt}\p{Nd}']*\b
+      \b[\p{Lu}\p{Lt}][\p{Ll}_\p{Lu}\p{Lt}\p{Nd}']*(?![\.''\w])
     name: constant.other.haskell
   qualifier:
     match: >-
-      \b[\p{Lu}\p{Lt}][\p{Ll}_\p{Lu}\p{Lt}\p{Nd}']*(?=\.)\b
-    name: meta.import.qualifier.haskell
+      \b[\p{Lu}\p{Lt}][\p{Ll}_\p{Lu}\p{Lt}\p{Nd}]*\.
+    name: entity.name.namespace.haskell
```
[the remark this answers is not in the document]
Comment (sheaf) [reply syntaxes/haskell.YAML-tmLanguage:531]: Oops I think I probably want to keep the apostrophe in there...

File: syntaxes/haskell.YAML-tmLanguage (hunk 1 above)
Comment (sheaf) [reply syntaxes/haskell.YAML-tmLanguage:364]: Yes, you can qualify symbolic operators too. See the test case `T0031`, all the qualifications therein are valid Haskell.

File: syntaxes/haskell.YAML-tmLanguage (hunk 2 above)
[the remark this answers is not in the document]
Comment (sheaf) [reply syntaxes/haskell.YAML-tmLanguage:531]: I added back the tick that I mistakenly removed here, and added a test case in `T0071b` (allowing ticks in qualified names).

File: syntaxes/haskell.YAML-tmLanguage [hunk 3]
```diff
@@ -76,7 +76,7 @@ patterns:
         |^(?!\1\s+\S|\s*$)    # at least one, no-further indented, non-whitespace character. I.e. a same-level declaration/implementation
     patterns:
       - include: '#deriving'
-      - begin: '^(\s*)(\b[\p{Lu}\p{Lt}][\p{Ll}_\p{Lu}\p{Lt}\p{Nd}'']*)\b'
+      - begin: '^(\s*)(\b[\p{Lu}\p{Lt}][\p{Ll}_\p{Lu}\p{Lt}\p{Nd}'']*)(?![\.''\w])'
```
[the remark this answers is not in the document]
Comment (sheaf) [reply syntaxes/haskell.YAML-tmLanguage:79]: I removed the lookahead and it doesn't seem to cause any issues. Pushed in new commit.

**Comment by sheaf** [comment]
I added many tricky edges cases and finally managed to fix them all, see the `Ticks` test case:
ticks_highlighting_examples

Note in particular that `T.'C` is not a qualified type, it should be parsed as:
 - `T` is a type,
 - `.` is a type operator,
 - `'C` is a promoted data constructor.

Let's see how GitHub does:

```haskell
  type T = '(F, (B,G.C,'[A,B,C]))
  type T = '() '[]
  type T = T.'C 
  type T = '(:+) '(AC.:+) 'AC.:++ 'AC.++
```
Not bad although it fails at qualified infix type operators.

**Pull request merged**